src/main/java/net/tropicraft/core/client/entity/model/ModelScubaGear.java

```diff
@@ -541,86 +548,100 @@ public void renderScubaGear(Entity entityIn, float limbSwing, float limbSwingAmo
+            Tank2m1.rotateAngleZ = 0F;
+            Tank2m1.renderWithRotation(scale);
+
+            Tank2m2.rotateAngleX = 0F;
+            Tank2m2.rotateAngleY = -1.570796F;
+            Tank2m2.rotateAngleZ = 0F;
+            Tank2m2.renderWithRotation(scale);
+
+            Tank2m3.rotateAngleX = 0F;
+            Tank2m3.rotateAngleY = -1.570796F;
+            Tank2m3.rotateAngleZ = 0F;
+            Tank2m3.renderWithRotation(scale);
+
+            Tank2m4.rotateAngleX = 0F;
+            Tank2m4.rotateAngleY = 0F;
+            Tank2m4.rotateAngleZ = 0F;
+            Tank2m4.renderWithRotation(scale);
+
+            Tank2m5.rotateAngleX = 0F;
+            Tank2m5.rotateAngleY = 0F;
+            Tank2m5.rotateAngleZ = 0F;
+            Tank2m5.renderWithRotation(scale);
+
+            Tank2m6.rotateAngleX = 0F;
+            Tank2m6.rotateAngleY = 0F;
+            Tank2m6.rotateAngleZ = 0F;
+            Tank2m6.renderWithRotation(scale);
+
+            Tank2m7.rotateAngleX = 0F;
+            Tank2m7.rotateAngleY = 0F;
+            Tank2m7.rotateAngleZ = 0F;
+            Tank2m7.renderWithRotation(scale);
+        }
+
+        if (gear.getTanks().getLeft() != null) {
+            ScubaMaterial material = gear.getStackInSlot(0).getItem() == ItemRegistry.pinkScubaTank ? ScubaMaterial.PINK : ScubaMaterial.YELLOW;
+            tm.bindTexture(ItemScubaGear.getArmorTexture(material));
+    
+            Tank1.rotateAngleX = 0F;
+            Tank1.rotateAngleY = 0F;
+            Tank1.rotateAngleZ = 0F;
+            Tank1.renderWithRotation(scale);
+    
+            Tank1m1.rotateAngleX = 0F;
+            Tank1m1.rotateAngleY = 0F;
+            Tank1m1.rotateAngleZ = 0F;
+            Tank1m1.renderWithRotation(scale);
+    
+            Tank1m2.rotateAngleX = 0F;
+            Tank1m2.rotateAngleY = -1.570796F;
+            Tank1m2.rotateAngleZ = 0F;
+            Tank1m2.renderWithRotation(scale);
+    
+            Tank1m3.rotateAngleX = 0F;
+            Tank1m3.rotateAngleY = -1.570796F;
+            Tank1m3.rotateAngleZ = 0F;
+            Tank1m3.renderWithRotation(scale);
+    
+            Tank1m4.rotateAngleX = 0F;
+            Tank1m4.rotateAngleY = 0F;
+            Tank1m4.rotateAngleZ = 0F;
+            Tank1m4.renderWithRotation(scale);
+    
+            Tank1m5.rotateAngleX = 0F;
+            Tank1m5.rotateAngleY = 0F;
+            Tank1m5.rotateAngleZ = 0F;
+            Tank1m5.renderWithRotation(scale);
+    
+            Tank1m6.rotateAngleX = 0F;
+            Tank1m6.rotateAngleY = 0F;
+            Tank1m6.rotateAngleZ = 0F;
+            Tank1m6.renderWithRotation(scale);
+    
+            Tank1m7.rotateAngleX = 0F;
+            Tank1m7.rotateAngleY = 0F;
+            Tank1m7.rotateAngleZ = 0F;
+            Tank1m7.renderWithRotation(scale);
+        }
+        Minecraft.getMinecraft().renderEngine.bindTexture(new ResourceLocation(((ItemArmor)gearStack.getItem()).getArmorTexture(gearStack, entityIn, EntityEquipmentSlot.CHEST, null)));
```

Comment:
wouldn't this be creating a new ResourceLocation every tick? Or is that necessary because of the entity?